Video-Transformers/README.md

```diff
@@ -0,0 +1,139 @@
+
+```objectivec
+@interface CustomTransformer : NSObject <OTCustomVideoTransformer>
+@end
+@implementation CustomTransformer
+- (void)transform:(nonnull OTVideoFrame *)videoFrame {
+    // Your custom transformation
+}
+@end
+```
+
+In this sample, to display one of the infinite transformations that can be applied to video frames, a logo is being added to the bottom right corner of the video.
+
+```objectivec
+@interface CustomTransformer : NSObject <OTCustomVideoTransformer>
+@end
+@implementation CustomTransformer
+- (void)transform:(nonnull OTVideoFrame *)videoFrame {
+    
+    UIImage* image = [UIImage imageNamed:@"Vonage_Logo.png"];
+
+    uint32_t videoWidth = videoFrame.format.imageWidth;
+    uint32_t videoHeight = videoFrame.format.imageHeight;
+
+    // Calculate the desired size of the image
+    CGFloat desiredWidth = videoWidth / 8;  // Adjust this value as needed
+    CGFloat desiredHeight = image.size.height * (desiredWidth / image.size.width);
+
+    // Resize the image to the desired size
+    UIImage *resizedImage = [self resizeImage:image toSize:CGSizeMake(desiredWidth, desiredHeight)];
+
+    // Get pointer to the Y plane
+    uint8_t* yPlane = [videoFrame getPlaneBinaryData:0];
+    
+    // Create a CGContext from the Y plane
+    CGContextRef context = CGBitmapContextCreate(yPlane, videoWidth, videoHeight, 8, videoWidth, CGColorSpaceCreateDeviceGray(), kCGImageAlphaNone);
+    
+    // Location of the image (in this case right bottom corner)
+    CGFloat x = videoWidth * 4/5;
+    CGFloat y = videoHeight * 1/5;
+    
+    // Draw the resized image on top of the Y plane
+    CGRect rect = CGRectMake(x, y, desiredWidth, desiredHeight);
+    CGContextDrawImage(context, rect, resizedImage.CGImage);
+    
+    CGContextRelease(context);
+}
+@end
+```
+
+Then set the `OTPublisherKit.videoTransformers` property to an array that includes the object that implements the
+OTCustomVideoTransformer interface:
+
+```objectivec
+CustomTransformer* logoTransformer;
+logoTransformer = [customTransformer alloc];
+OTVideoTransformer *myCustomTransformer = [[OTVideoTransformer alloc] initWithName:@"logo" transformer:logoTransformer];
+
+NSMutableArray * myVideoTransformers = [[NSMutableArray alloc] init];
+[myVideoTransformers addObject:myCustomTransformer];
+
+_publisher.videoTransformers = [[NSArray alloc] initWithArray:myVideoTransformers];
+```
+
+You can combine the Vonage Media library transformer (see the previous section) with custom transformers or apply
+multiple custom transformers by adding multiple PublisherKit.VideoTransformer objects to the ArrayList used
+for the `OTPublisherKit.videoTransformers` property.
+
+## Clearing video transformers for a publisher
+
+To clear video transformers for a publisher, set the `OTPublisherKit.videoTransformers` property to an empty array.
+
+```objectivec
+_publisher.videoTransformers = [[NSArray alloc] init];
```

Comment:
If you have time, a toggle on Blur showing clear transformers, would be cool in the sample.. if you want..

Reply:
Yes, of course. The more the better. Would you be ok having a button which would toggle the transformers?